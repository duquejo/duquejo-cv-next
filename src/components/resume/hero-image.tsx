import Image from 'next/image';
import { cn } from '@/lib';
import { Circle, Gamepad, LoaderCircle, Music, Send } from 'lucide-react';
import avatarPicture from '../../../public/static/img/avatar_410.png';

interface Props {
  className?: string;
}

export const HeroImage = ({ className }: Props) => {
  return (
    <div data-testid="hero-image" className={cn('overflow-hidden relative', className)}>
      <Music
        className="absolute top-0 text-primary/50 animate-bounce-subtle repeat-infinite z-10"
        strokeWidth="3"
        size={54}
      />
      <Gamepad
        className="absolute bottom-0 left-7 text-primary/50 animate-wiggle z-10"
        strokeWidth="3"
        size={60}
      />
      <LoaderCircle
        className="absolute right-10 text-primary animate-spin duration-slow repeat-infinite z-10"
        strokeWidth="5"
        size={44}
      />
      <Send
        className="absolute right-8 bottom-5 text-primary animate-bounce-subtle repeat-infinite z-10"
        strokeWidth="3"
        size={50}
      />

      {/*Right*/}
      <Circle
        className="absolute right-36 top-10 text-primary/50 animate-pulse duration-slow"
        strokeWidth="5"
        size={40}
      />
      {/*Left*/}
      <Circle
        className="absolute left-20 bottom-24 text-primary/50 animate-pulse duration-slow delay-150"
        strokeWidth="5"
        size={20}
      />
      <picture className="relative">
        <Image
          className="rounded-full backdrop-blur-sm bg-primary/10 shadow-inner"
          alt="JosÃ© Miguel Duque"
          height={410}
          width={410}
          priority={true}
          quality={100}
          src={avatarPicture}
        />
      </picture>
    </div>
  );
};
