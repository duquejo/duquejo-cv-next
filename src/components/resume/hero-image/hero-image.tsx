import Image from 'next/image';
import { cn } from '@/lib';
import { Circle, Gamepad, LoaderCircle, Music, Send } from 'lucide-react';

interface Props {
  className?: string;
}

export const HeroImage = ({ className }: Props) => {
  return (
    <div data-testid="hero-image" className={cn('overflow-hidden relative', className)}>
      <Music
        className="absolute top-0 text-primary/50 animate-bounce-subtle repeat-infinite z-10"
        strokeWidth="3"
        size={64}
      />
      <Gamepad
        className="absolute bottom-0 left-7 text-primary/50 animate-wiggle z-10"
        strokeWidth="3"
        size={70}
      />
      <LoaderCircle
        className="absolute right-10 text-primary animate-spin duration-slow repeat-infinite z-10"
        strokeWidth="5"
        size={64}
      />
      <Send
        className="absolute right-8 bottom-5 text-primary animate-bounce-subtle repeat-infinite z-10"
        strokeWidth="3"
        size={50}
      />

      {/*Right*/}
      <Circle
        className="absolute right-36 top-10 text-primary/50 animate-pulse duration-slow"
        strokeWidth="5"
        size={40}
      />
      {/*Left*/}
      <Circle
        className="absolute left-20 bottom-24 text-primary/50 animate-pulse duration-slow delay-150"
        strokeWidth="5"
        size={20}
      />
      <picture className="object-cover">
        <Image
          className="rounded-full backdrop-blur-sm bg-primary/10 shadow-inner"
          alt="JosÃ© Miguel Duque"
          width="950"
          height="950"
          priority
          src="/static/img/avatar_cropped_950.png"
        />
      </picture>
    </div>
  );
};
