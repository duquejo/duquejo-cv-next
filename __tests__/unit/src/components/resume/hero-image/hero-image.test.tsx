import { HeroImage } from '@/components/resume/hero-image/hero-image';
import { render, screen } from '@testing-library/react';

describe('<HeroImage /> tests', () => {
  it('Should match the snapshot with the default args', () => {
    const { container } = render(<HeroImage />);

    expect(container).toMatchSnapshot();
  });

  it('Should add the given css classes if they are given', () => {
    render(<HeroImage className="bg-red-500" />);

    expect(screen.getByTestId('hero-image')).toBeInTheDocument();
    expect(screen.getByTestId('hero-image')).toHaveClass('bg-red-500');
  });
});
