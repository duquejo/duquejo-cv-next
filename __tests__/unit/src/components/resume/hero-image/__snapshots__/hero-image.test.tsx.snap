// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<HeroImage /> tests > Should match the snapshot with the default args 1`] = `
<div>
  <div
    class="overflow-hidden relative"
    data-testid="hero-image"
  >
    <picture
      class="relative"
    >
      <img
        alt="JosÃ© Miguel Duque"
        class="rounded-full backdrop-blur-xs bg-primary/10 shadow-inner"
        data-nimg="1"
        decoding="async"
        height="410"
        src="/_next/image?url=%2Fpublic%2Fstatic%2Fimg%2Favatar_410.png&w=828&q=100"
        srcset="/_next/image?url=%2Fpublic%2Fstatic%2Fimg%2Favatar_410.png&w=640&q=100 1x, /_next/image?url=%2Fpublic%2Fstatic%2Fimg%2Favatar_410.png&w=828&q=100 2x"
        style="color: transparent;"
        width="410"
      />
    </picture>
  </div>
</div>
`;
